pkg/operation/botanist/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -191,18 +192,24 @@ type Data struct {
 
 // Deploy uses the client to create or update the OperatingSystemConfig custom resources.
 func (o *operatingSystemConfig) Deploy(ctx context.Context) error {
-	return o.reconcile(ctx, func(d deployer) error {
+	if err := o.reconcile(ctx, func(d deployer) error {
 		_, err := d.deploy(ctx, v1beta1constants.GardenerOperationReconcile)
 		return err
-	})
+	}); err != nil {
+		return gardencorev1beta1helper.DeprecatedDetermineError(err)
+	}
+	return nil
 }
 
 // Restore uses the seed client and the ShootState to create the OperatingSystemConfig custom resources in the Shoot
 // namespace in the Seed and restore its state.
 func (o *operatingSystemConfig) Restore(ctx context.Context, shootState *v1alpha1.ShootState) error {
-	return o.reconcile(ctx, func(d deployer) error {
+	if err := o.reconcile(ctx, func(d deployer) error {
 		return extensions.RestoreExtensionWithDeployFunction(ctx, o.client, shootState, extensionsv1alpha1.OperatingSystemConfigResource, d.deploy)
-	})
+	}); err != nil {
+		return gardencorev1beta1helper.DeprecatedDetermineError(err)
+	}
+	return nil
```

Comment:
We can remove this from `Restore` functions as well like `Deploy` and have it only in the "wait" functions.